x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -662,11 +662,21 @@ private static void checkGroupingFunctionTarget(GroupingFunction f, Set<Failure>
     private static void checkFilterOnAggs(LogicalPlan p, Set<Failure> localFailures, AttributeMap<Expression> attributeRefs) {
         if (p instanceof Filter) {
             Filter filter = (Filter) p;
-            if ((filter.child() instanceof Aggregate) == false) {
+            LogicalPlan filterChild = filter.child();
+            if (filterChild instanceof Aggregate == false) {
                 filter.condition().forEachDown(Expression.class, e -> {
                     if (Functions.isAggregate(attributeRefs.getOrDefault(e, e))) {
-                        localFailures.add(
-                            fail(e, "Cannot use WHERE filtering on aggregate function [{}], use HAVING instead", Expressions.name(e)));
+                        if (filterChild instanceof Project) {
+                            filter.condition().forEachDown(FieldAttribute.class,
+                                fa -> localFailures.add(
```

Comment:
Slight adjustment " or in an aggregate function" (drop `be used`)

Reply:
Adjusted.